From e42ec234e180e489ef898ce05c99d9807737bc68 Mon Sep 17 00:00:00 2001
From: Nariman Poushin <nariman@opensource.wolfsonmicro.com>
Date: Thu, 13 Feb 2014 02:09:31 +0000
Subject: [PATCH 128/351] switch: arizona: Use high z for hpdet_cb to notify
 no headset

Use out of spec impedance to report open circuit on the headphone
to the hpdet_cb.

Change-Id: I3b12f53e6ff56a1b86921bb15a32fd1c0b4724b0
Signed-off-by: Nariman Poushin <nariman@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c   |    5 +++++
 include/linux/mfd/arizona/pdata.h |    5 +++++
 2 files changed, 10 insertions(+)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 6160790..e6cf898 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -1260,6 +1260,11 @@ static irqreturn_t arizona_jackdet(int irq, void *data)
 				   ARIZONA_JACK_DETECT_DEBOUNCE,
 				   ARIZONA_MICD_CLAMP_DB | ARIZONA_JD1_DB,
 				   ARIZONA_MICD_CLAMP_DB | ARIZONA_JD1_DB);
+
+		/* Use a sufficiently large number to indicate open circuit */
+		if (arizona->pdata.hpdet_cb) {
+			arizona->pdata.hpdet_cb(ARIZONA_HP_Z_OPEN);
+		}
 	}
 
 	if (arizona->pdata.micd_timeout)
diff --git a/include/linux/mfd/arizona/pdata.h b/include/linux/mfd/arizona/pdata.h
index f9c61d6..8ac525f 100644
--- a/include/linux/mfd/arizona/pdata.h
+++ b/include/linux/mfd/arizona/pdata.h
@@ -8,6 +8,8 @@
  * published by the Free Software Foundation.
  */
 
+#include <linux/kernel.h>
+
 #ifndef _ARIZONA_PDATA_H
 #define _ARIZONA_PDATA_H
 
@@ -71,6 +73,9 @@
 
 #define ARIZONA_MAX_PDM_SPK 2
 
+/* Treat INT_MAX impedance as open circuit */
+#define ARIZONA_HP_Z_OPEN INT_MAX
+
 struct regulator_init_data;
 
 struct arizona_micbias {
-- 
1.7.9.5

